<a href="https://colab.research.google.com/github/HwangHanJae/ml-definitive-guide-pratice/blob/main/pandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

read_csv()의 특징 :
- 필드 구분 문자가 콤마(',')로 구분
- (파일명, sep ='\t')으로 탭으로 구분된 데이터 로드 가능

read_table()의 특징 : 
- 필드 구분 문자가 탭('\t')으로 구분

read_fwf()의 특징 : 
- Fixed Width 고정 길이 기반의 칼럼 포맷

In [3]:
import pandas as pd

#캐글 - 타이타닉 탑승자 데이터
#read_csv()로 읽어오기, (filepath_of_buffer, sep = ','...)
#filepath_or_buffer 자리에는 파일 경로

file_path = "/content/drive/MyDrive/Colab Notebooks/DataScience/Data/titanic/train.csv"
titanic_df = pd.read_csv(file_path)
print("titalnic_df 변수 type : ", type(titanic_df))

#head()는 defalut는 5, 인자값으로 n을 지정하면 n만큼 n개의 로우를 반환
titanic_df.head()
#index(0,1,3,4...)객체는 sql의 pk와 유사하게 고유의 레코드를 식별하는 역할

titalnic_df 변수 type :  <class 'pandas.core.frame.DataFrame'>


PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0            1         0       3  ...   7.2500   NaN         S
1            2         1       1  ...  71.2833   C85         C
2            3         1       3  ...   7.9250   NaN         S
3            4         1       1  ...  53.1000  C123         S
4            5         0       3  ...   8.0500   NaN         S

[5 rows x 12 columns]

In [7]:
#shape : 생성된 DataFrame 객체의 크기를 확인

print("titanic_df 의 크기 : ",titanic_df.shape)
print("생성된 titanic_df는 {}개의 행(로우)과 {}개의 열(칼럼)을 가지고 있습니다.".format(titanic_df.shape[0], titanic_df.shape[1]))

titanic_df 의 크기 :  (891, 12)
생성된 titanic_df는 891개의 행(로우)과 12개의 열(칼럼)을 가지고 있습니다.


In [8]:
#info() : 총 데이터 건수와 데이터타입, Null 건수를 확인
titanic_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


info() 해석
1. RangeIndex는 DataFrame index의 범위를 나타냅니다. 
- 전체 로우의 수를 알 수 있습니다. 데이터는 891개 로우 입니다. 그리고 컬럼수는 12개입니다.

2. Dtype은 칼럼별 타입을 나타냅니다.
- Pclass 칼럼의 데이터 타입은 int64형 입니다.
- Name 칼럼의 데이터 타입은 object입니다. 이는 문자열타입으로 생각해도 무방합니다.

3. Non-Null Count를 보면 몇개의 데이터가 non-null인지 나타냅니다.
- Age 칼럼의 714 non-null의 의미는 Age 칼럼 891개 데이터 중 714개가 Null이 아니며 177개는 Null이라는 의미입니다.

4. dtype : 은 전체 12개의 칼럼들의 타입을 요약한 것 입니다.
- 2개 칼럼이 float64, 5개의 칼럼이 int64, 5개 칼럼이 object 타입입니다.


In [10]:
#describe() 오직 숫자형(int, float 등)칼럼의 분포도만 조사하며 자동으로 object 타입의 칼럼은 출력에서 제외합니다.
titanic_df.describe()

PassengerId    Survived      Pclass  ...       SibSp       Parch        Fare
count   891.000000  891.000000  891.000000  ...  891.000000  891.000000  891.000000
mean    446.000000    0.383838    2.308642  ...    0.523008    0.381594   32.204208
std     257.353842    0.486592    0.836071  ...    1.102743    0.806057   49.693429
min       1.000000    0.000000    1.000000  ...    0.000000    0.000000    0.000000
25%     223.500000    0.000000    2.000000  ...    0.000000    0.000000    7.910400
50%     446.000000    0.000000    3.000000  ...    0.000000    0.000000   14.454200
75%     668.500000    1.000000    3.000000  ...    1.000000    0.000000   31.000000
max     891.000000    1.000000    3.000000  ...    8.000000    6.000000  512.329200

[8 rows x 7 columns]

In [12]:

value_counts = titanic_df["Pclass"].value_counts()

print(value_counts)

3    491
1    216
2    184
Name: Pclass, dtype: int64


In [13]:
#DataFrame의 []연산자 내부에 칼럼명을 입력하면 해당 칼럼에 해당하는 Series 객체를 반환
titanic_pclass = titanic_df["Pclass"]

print(type(titanic_pclass))

<class 'pandas.core.series.Series'>


In [15]:
#Series 타입에도 head() 메서드를 사용할 수 있음
titanic_pclass.head()

0    3
1    1
2    3
3    1
4    3
Name: Pclass, dtype: int64